spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelFactory.java

```diff
@@ -25,5 +26,24 @@
  */
 public interface ChannelFactory {
 
-	MessageChannel createChannel(String name, Class<?> channelType) throws Exception;
+	/**
+	 * Create a {@link SubscribableChannel} that will be bound to the message channel
+	 * {@link org.springframework.cloud.stream.binder.Binder}.
+	 *
+	 * @param name name of the message channel
+	 * @return Subscribable message channel
+	 * @throws Exception
+	 */
+	SubscribableChannel createSubscribableChannel(String name) throws Exception;
```

Comment:
why do these 2 methods have `throws Exception`? I don't think it's necessary (and if if were we would want to reconsider throwing *checked* Exceptions)